port/win/port_win.h

```diff
@@ -81,8 +81,8 @@ namespace rocksdb {
 
 namespace port {
 
-// VS 15
-#if (defined _MSC_VER) && (_MSC_VER >= 1900)
+// VS 15 / MinGW
+#if !defined(_MSC_VER) || (_MSC_VER >= 1900)
```

Comment:
Moreover, I don't think there's a reason to exclude clang or ICC for example. Do they define __MINGW32__?